contracts/AirDrop.sol

```diff
@@ -0,0 +1,186 @@
+pragma solidity 0.6.4;
+
+import "./interface/IBEP20.sol";
+import "./interface/ITokenHub.sol";
+import "./interface/IAirDrop.sol";
+import "./interface/IParamSubscriber.sol";
+import "./lib/SafeMath.sol";
+import "./lib/BytesToTypes.sol";
+import "./System.sol";
+import "./MerkleProof.sol";
+
+contract AirDrop is IAirDrop, IParamSubscriber, System {
+    using SafeMath for uint256;
+
+    string public constant sourceChainID = "Binance-Chain-Ganges";
+    address public approvalAddress = 0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa;
+    bytes32 public override merkleRoot = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    bool public merkleRootAlreadyInit = false;
+
+    // claimedMap is used to record the claimed token.
+    mapping(bytes32 => bool) private claimedMap;
+
+    function isClaimed(bytes32 node) public view override returns (bool) {
+        return claimedMap[node];
+    }
+
+    function claim(
+        bytes32 tokenSymbol, uint256 amount,
+        bytes calldata ownerPubKey, bytes calldata ownerSignature, bytes calldata approvalSignature,
+        bytes32[] calldata merkleProof) external override {
+        // Recover the owner address and check signature.
+        bytes memory ownerAddr = _verifyTMSignature(ownerPubKey, ownerSignature, _tmSignatureHash(tokenSymbol, amount, msg.sender));
+        // Generate the leaf node of merkle tree.
+        bytes32 node = keccak256(abi.encodePacked(ownerAddr, tokenSymbol, amount));
+    
+        // Check if the token is claimed.
+        require(isClaimed(node), "AlreadyClaimed");
+
+        address contractAddr = address(0x00);
+        // Check if the token is exist.
+        if (tokenSymbol != "BNB") {
```

Comment:
can they compare directly.

 ` bytes32 constant public BEP2_TOKEN_SYMBOL_FOR_BNB = 0x424E420000000000000000000000000000000000000000000000000000000000; // "BNB"`

Reply:
fixed

Reply:
Because the current implementation will need a cross contract call, I will suggest we have a const just in this smart contract.